src/core/sock/sockinfo_tcp.cpp

```diff
@@ -2719,24 +2735,20 @@ int sockinfo_tcp::bind(const sockaddr *__addr, socklen_t __addrlen)
 
     int ret = 0;
 
-    si_tcp_logdbg("to %s", sockaddr2str(__addr, __addrlen, true).c_str());
+    si_tcp_logdbg("to %s, m_bind_no_port=%d", sockaddr2str(__addr, __addrlen, true).c_str(),
+                  m_bind_no_port);
 
-    if (m_sock_state == TCP_SOCK_BOUND) {
-        si_tcp_logfuncall("already bounded");
-        errno = EINVAL;
-        return -1;
-    }
-
-    if (m_sock_state != TCP_SOCK_INITED) {
+    if (m_sock_state > TCP_SOCK_BOUND) {
```

Comment:
Shouldn't it be `m_sock_state >= TCP_SOCK_BOUND`?

Reply:
yes. done.